hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -387,6 +388,19 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 				ObservedGeneration: nodePool.Generation,
 			})
 		}
+	} else if nodePool.Spec.Platform.Type == hyperv1.PowerVSPlatform {
+		powervsImage, _, err := getPowerVSImage(hcluster.Spec.Platform.PowerVS.Region, releaseImage)
+		if err != nil {
+			setStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+				Type:               hyperv1.NodePoolValidAMIConditionType,
+				Status:             corev1.ConditionFalse,
+				Reason:             hyperv1.NodePoolValidationFailedConditionReason,
+				Message:            fmt.Sprintf("Couldn't discover an PowerVS Image for release image %q: %s", nodePool.Spec.Release.Image, err.Error()),
+				ObservedGeneration: nodePool.Generation,
+			})
+			return ctrl.Result{}, fmt.Errorf("couldn't discover an AMI for release image: %w", err)
+		}
+		ami = powervsImage.Release
```

Comment:
Does the AMI term has a meaning in PowerVS? Or this just copied from AWS?

Reply:
I just wanted to reuse the variable here so that I can use that in the [machineTemplateBuilders](https://github.com/openshift/hypershift/blob/f4ef154b9e034311b3db0a43d961d8b0e23dff3f/hypershift-operator/controllers/nodepool/nodepool_controller.go#L1474), may be I need find a better way for this one.

Note: not really comfortable seeing provider based params to that function because easily you may end up adding many more in the future based on the additional providers we add in the future.